www/js/app.js

```diff
@@ -102,5 +103,23 @@ angular.module('emission', ['ionic',
   // alert("about to fall back to otherwise");
   // if none of the above states are matched, use this as the fallback
   $urlRouterProvider.otherwise('/splash');
+
+  // Allow the use of MessageForm interpolation for Gender and Plural.
+  $translateProvider.addInterpolation('$translateMessageFormatInterpolation');
+
+  // Define where we can find the .json and the fallback language
+  $translateProvider
+    .fallbackLanguage('en')
+    .registerAvailableLanguageKeys(['en', 'fr'], {
+      'en_*': 'en',
```

Comment:
so as we add translations from translatewiki, we would add the list here?
As a future enhancement, is it possible to determine this dynamically by iterating through the files in `www/i18n`?

Reply:
As we determined the language by accessing browser properties, It seemed necessary to "filter" the locales. For instance, a browser will return `en_US` or `en_UK` which could be seen as the same language and will try to read `en_US.json` or `en_UK.json`. When I register the languages keys, I avoid this behaviour. [Language Negotiation](https://angular-translate.github.io/docs/#/guide/09_language-negotiation)